drivers/i2c/i2c_nrfx_twi.c

```diff
@@ -157,28 +163,34 @@ static int init_twi(struct device *dev)
 static int twi_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
 				void *context, device_pm_cb cb, void *arg)
 {
-	static u32_t current_state = DEVICE_PM_ACTIVE_STATE;
-
 	if (ctrl_command == DEVICE_PM_SET_POWER_STATE) {
 		u32_t new_state = *((const u32_t *)context);
 
-		if (new_state != current_state) {
+		if (new_state != get_dev_data(dev)->pm_state) {
 			switch (new_state) {
 			case DEVICE_PM_ACTIVE_STATE:
 				init_twi(dev);
-				i2c_nrfx_twi_configure(
-					dev, get_dev_data(dev)->dev_config);
+				if (get_dev_data(dev)->dev_config) {
+					i2c_nrfx_twi_configure(
+						dev,
+						get_dev_data(dev)->dev_config);
+				}
 				break;
 
-			default:
+			case DEVICE_PM_LOW_POWER_STATE:
+			case DEVICE_PM_SUSPEND_STATE:
+			case DEVICE_PM_OFF_STATE:
 				nrfx_twi_uninit(&get_dev_config(dev)->twi);
 				break;
+
+			default:
+				return -ENOTSUP;
```

Comment:
The value needs to be passed as the second parameter to the `cb` function. At least that's what other drivers do. This parameter is not well/at all documented in `device.h` (its name just says `status`), so I think we should just follow the others.

Reply:
Maybe you could use the pattern from [the PWM driver](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/pwm/pwm_nrfx.c#L322)?